flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -205,6 +212,10 @@ public void run() {
 					if (jobArchives == null) {
 						continue;
 					}
+
+					Arrays.sort(jobArchives, Comparator.comparingLong(FileStatus::getModificationTime).reversed());
```

Comment:
It should be documented that the `maxHistorySize` is enforced per archive directory, and not globally.

Reply:
Done.